dataflux_pytorch/dataflux_mapstyle_dataset.py

```diff
@@ -200,7 +200,8 @@ def _list_GCS_blobs_with_retry(self):
                     prefix=self.config.prefix,
                     retry_config=self.config.list_retry_config,
                 )
-                lister.client = self._get_or_create_storage_client()
+                if self.storage_client is not None:
+                    lister.client = self.storage_client
```

Comment:
Thanks for noticing and fixing this! A few questions:
1. Are you sure this is the only place in the code that is affected? Will there be a similar error for the download parts of the code?
2. I thought when this was fixed before we had added tests that would have caught this. Is there any additional test we can add that would help prevent this from being reintroduced?

Reply:
Adding storage_client to DataFluxMapStyleDataset object was causing pickling error as it was used by pytorch dataloader. pytorch dataloader calls multiprocessing.  With _get_or_create_storage_client method, the client is not added to DataFluxMapStyleDataset object. 

In dataflux_mapstyle_dataset.py we are calling download_single and threaded_download which does not call multiprocessing.  Multiprocessing is called only by parallel_composed_download.

Also ran a test and looks like this was the only place it will fail. 

Our unit test wasn't covering listing,  I'll add more unit tests to cover this.

Reply:
Thanks for catching this! I misunderstood where the pickling error was being thrown. This looks like a good solution. I added a suggestion for additional comment to make the issue clear.